### ISSUE TITLE
BCC not working from https://app.sendgrid.com/settings/mail_settings

### ISSUE THREAD

**Issue by cfloydcorelogic** [sendgrid/sendgrid-csharp#534]
#### Issue Summary

When I add an email address to the BCC in the https://app.sendgrid.com/settings/mail_settings it does not send an email to the BCC.

#### Steps to Reproduce

1. Add BCC email
2. Send email post through API v3
3. Email is sent to primary
3. Email is not sent to BCC

Any other information you want to share that is relevant to the issue being reported. Especially, why do you consider this to be a bug? The SendGrid support rep told me that I should post here. They said that everything we are doing looks correct.

What do you expect to happen instead? BCC the email address in the BCC field in the SendGrid mail settings.

#### Technical details:

* sendgrid-nuget Version: 9.9.0
* Platform .NET Core 2.0
* Sendgrid API version 3

```csharp
            var client = new SendGridClient(_apiKey);
            var sendGridFrom = new SendGrid.Helpers.Mail.EmailAddress(from.Email, from.Alias);
            var sendGridTo = new SendGrid.Helpers.Mail.EmailAddress(to.Email, to.Alias);
            var msg = MailHelper.CreateSingleEmail(sendGridFrom, sendGridTo, subject, stringContent, htmlContent);
            var MailSettings = new MailSettings();
            MailSettings.SandboxMode = new SandboxMode();
            MailSettings.SandboxMode.Enable = _isSandbox;
            msg.MailSettings = MailSettings;
            Response response = await client.SendEmailAsync(msg);
            if(((int)response.StatusCode) >= 400) throw new Exception(await response.Body.ReadAsStringAsync());
```

**Comment by cfloydcorelogic** [comment]
| vsg    Oct 3, 15:16 MDT    Chat transcript:   Visitor: I have the mail settings set to always BCC a copy of the email, but   the email isn't receiving them   Visitor: that is what you said last time before you disconnected me :)   Poul P.: Hello, thanks for chatting in. Apologies for the delayed response.   Poul P.: One moment please while I take a look at the event logs to find out   why the BCC email isn't receiving emails.   Poul P.: I'm going to try sending a test email through the account to check   if it sends a BCC to you.   Visitor: ok   Visitor: Nothing in Gmail yet, let me know when you want me to check.   Poul P.: Okay you should see something shortly, I'm seeing that the message   had been sent to both the test email and the CC address cfloyd@corelogic.com   Poul P.: Are you sending the message via SMTP, API v2 or API v3?   Visitor: I got the test in gmail, waiting for CoreLogic to get it   Visitor: API v3   Visitor: my BCC account is gmail   Visitor: still nothing in corelogic   Poul P.: My apologies I mean the gmail account that was CC'd I got it mixed   up, my apologies   Poul P.: Since you received it in your BCC account then it did work properly.   Poul P.: In the API request are you adding a mail_settings object?   Visitor: I am not adding it in the API, I am using it in the mail settings   from the site   Visitor: https://app.sendgrid.com/settings/mail_settings   Poul P.: I understand, please send a message via API to sendgridtesting@gmail.com. I'll   check the email events on the account to find out if it's sent to you as   well.   Visitor: ok, one seic   Visitor: sec   Poul P.: No worries, I'll be here.   Visitor: you should be getting it soon   Poul P.: Thanks, I'll take a look.   Poul P.: For some reason it's not CCing that even though it is being sent   through the account. Please provide me with the API call that you had made to   sendgridtesting@gmail.com   Visitor: we will have to do it again because I have to get a developer to   turn on debugging   Poul P.: Ah okay, do you have any insight into how the developer's sending   the message?   Poul P.: For some reason it's not BCCing when sending via the API but through   a test with me it is which makes me think there might be some issue with the   way the email request is being made.   Visitor: just that they are using the sengrid API    Visitor: He is sitting behind me.   Visitor: One sec, we are going to send another one to you with debugging   Poul P.: Alright sounds good, I'll be here.   Visitor: this is how we are building it    Visitor: var client = new SendGridClient(_apiKey); var sendGridFrom = new   SendGrid.Helpers.Mail.EmailAddress(from.Email, from.Alias); var sendGridTo =   new SendGrid.Helpers.Mail.EmailAddress(to.Email, to.Alias); var msg =   MailHelper.CreateSingleEmail(sendGridFrom, sendGridTo, subject,   stringContent, htmlContent); var MailSettings = new MailSettings();   MailSettings.SandboxMode = new SandboxMode(); MailSettings.SandboxMode.Enable   = _isSandbox; msg.MailSettings = MailSettings; Response response = await   client.SendEmailAsync(msg); if(((int)response.StatusCode) >= 400) throw   new Exception(await response.Body.ReadAsStringAsync());   Visitor: isSandbox would be false in the site I am emailing you from   Poul P.: Alright hmmm, Would it be possible to not apply the mailsettings to   the msg?   Visitor: sure   Poul P.: This might be a bit of a stretch but I'm wondering if applying the   mailsettings object is making the BCC settings false. I'm going out on a limb   but it could be that it's initialized as false.   Visitor: we just sent one to you   Visitor: without mail settings   Poul P.: Great, thanks. One moment please.   Poul P.: It's still not working for some reason.Lets try this, could you   change the BCC address to sendgridtesting@gmail.com.   Then send another message via the API just make sure not to send it to sendgridtesting@gmail.com   Poul P.: Also just to confirm did you have the debug set up for that last   message?   Visitor: we didn't have debug on the last test, we just made the change to   not use mail settings   Visitor: changing the bcc now    Poul P.: Ah okay, after making the change to the BCC lets try it with the   debug mode just to double check. I'm at a bit of a loss for why this is   occurring but changing the BCC was at the suggestion of my colleague.   Visitor: debug mode just steps through the send process, once we get to this   line Response response = await client.SendEmailAsync(msg); we can't see what   happens   Poul P.: Ah okay.   Poul P.: Let me know once you've sent that message and we'll take a look if   it's being applied.   Visitor: just sent it   Visitor: waiting on my email to cfloyd@corelogic.com   Visitor: you are the BCC   Visitor: will let you know when I have it   Visitor: just got it    Poul P.: Okay, it wasn't sent to the BCC neither. The only other two things I   can think to do is I would first try updating to the latest version just in   case this resolves the issue. I see you're using 9.8.0 and it's currently   9.9.0 (although there isn't anything about this in the changelog). The other   thing I would do is reach out to Community Dev team by creating an issue at   our github page here: https://github.com/sendgrid/sendgrid-csharp/issues   They would have much more insight into how the helper works in the background   and would be able to provide further guidance in case their is some issue   within our API Library.   Visitor: ok   Poul P.: Apologies that I can't be of more assistance for why this isn't   working through our library. From what you've provided and the settings I   can't see anything that would be causing this to not send to the BCC.   Visitor: ok, I will post to git
-- | --

**Comment by thinkingserious** [comment]
Hi @cfloydcorelogic,

Thank you for taking the time to share your issue with us. I have added this to our backlog for review. 

In the mean time, you might want to try sending the email directly using our [interactive docs](https://sendgrid.com/docs/API_Reference/api_v3.html) or cURL to help determine whether this is a library issue.

With Best Regards,

Elmer

**Comment by Niladri24dutta** [comment]
@cfloydcorelogic  in your code i am not able to see any `bcc` property have you tried using `AddBcc` method of the library?

**Comment by cfloydcorelogic** [comment]
I am not doing it from the code.

I am going it from the SendGrid site.

Login to SendGrid
Click settings
Click mail settings
Add email address to BCC setting
![image](https://user-images.githubusercontent.com/30604843/31180163-822de8c4-a8e3-11e7-8000-8efed9729e2a.png)

**Comment by mbernier** [comment]
@cfloydcorelogic can you please try this from our [interactive docs](https://sendgrid.com/docs/API_Reference/api_v3.html), to see if it is something we are doing with your account or if it is the library itself?

**Comment by mbernier** [comment]
Oops, didn't mean to click close.

**Comment by cfloydcorelogic** [comment]
Using the interactive docs, I receive the email.  
I just want to be clear. Our application that we developed can send email using SendGrid with no issues.  
We can even send BCC emails with no problems. 
What I can't get to work is from the SendGrid webpage.  There is a BCC option that allows you to BCC every email you send to the BCC field.

**Comment by thinkingserious** [comment]
Hi @cfloydcorelogic,

When you use the interactive docs does the BCC option that allows you to BCC every email you send to the BCC field work?

**Comment by cfloydcorelogic** [comment]
Yes, when I post from the interactive docs, the SendGrid BCC works.

**Comment by thinkingserious** [comment]
Thanks @cfloydcorelogic,

We've added this to our backlog for further investigation.

**Comment by cfloydcorelogic** [comment]
For some reason, I have started to receive TXT messages from the SendGrid BCC now.

**Comment by thinkingserious** [comment]
Hmm, that is very strange @cfloydcorelogic,

Could you please report that issue to our support team so that they can look deeper to your account and troubleshoot?

**Comment by sthomas56** [comment]
@cfloydcorelogic  @thinkingserious @mbernier 

I am having this same issue where I have the BCC setting activated in SendGrid, but it is not sending it to the address I specified. Was a solution ever found?

**Comment by cfloydcorelogic** [comment]
Nothing from the support team yet.

**Comment by sthomas56** [comment]
@cfloydcorelogic 
Really this was 2 years ago and there is still no solution? So I am assuming this won't be fixed anytime soon?

**Comment by sthomas56** [comment]
@cfloydcorelogic 

Damn this was 2 years ago and still no solution haha. Did you find a workaround for it using the interactive docs? No matter what I try, when I send bulk emails, it only BCCs one out of the batch

**Comment by cfloydcorelogic** [comment]
I didn't pursue it. The guy who was wanting the BCC's decided that he no longer needed them.

**Comment by sthomas56** [comment]
@cfloydcorelogic Ok thanks for getting back to me.

**Comment by childish-sambino** [comment]
I've not been able to recreate the issue locally so I don't see this being an issue with the helper library itself. For that reason I'm closing it out.

If you're still running into this issue, the best thing to do is [file a support ticket](https://support.sendgrid.com) so that they can debug the issue on the backend.

**Comment by thinkingserious** [comment]
Hello @cfloydcorelogic,

Thanks for submitting a GitHub issue! We are very sorry that you are running into this problem. In order to better serve you, as this does not present itself as a library specific issue, we would like to ask that you reach out to our support team at <a href="https://support.sendgrid.com">https://support.sendgrid.com</a>.

Thank you!

SendGrid DX Team
